### PR TITLE
core/chains: wip metis config; rename Optimism2 -> L2Suggested gas estimator

### PR DESCRIPTION
**Pull request by jmank88**
Add defaults for metis test and mainnet.

- [x] start with optimism defaults
- [x] rename `Optimism2` gas estimator to `L2Suggested`
- [x] remove `Optimism` OVM 1.0 gas estimator
- [ ] differences from optimism? (config fields, new client errors to map, etc.) https://github.com/MetisProtocol/mvm/blob/develop/l2geth/core/tx_pool.go#L58-L91
- [x] bump libocr https://app.shortcut.com/chainlinklabs/story/39520/add-support-to-libocr-for-a-few-networks
- [x] changelog

### CODE REVIEW

File: core/chains/evm/config/chain_specific_config.go [hunk 1]
```diff
@@ -335,6 +335,23 @@ func setChainSpecificConfigDefaultSets() {
...
+	metisMainnet.gasBumpThreshold = 0 // Never bump gas on metis
+	metisMainnet.gasEstimatorMode = "L2Suggested"
+	metisMainnet.linkContractAddress = "TODO"
+	metisMainnet.minIncomingConfirmations = 1
+	metisMainnet.minGasPriceWei = *big.NewInt(0) // Metis uses the L2Suggested estimator; we don't want to place any limits on the minimum gas price
+	metisMainnet.minRequiredOutgoingConfirmations = 0
+	metisMainnet.ocrContractConfirmations = 1
+	//TODO more?
+	metisRinkeby := metisMainnet
+	metisRinkeby.linkContractAddress = "TODO"
```
Comment (jmank88) [inline core/chains/evm/config/chain_specific_config.go:353]: This is the interesting part. I started with the settings from Optimism that made sense to me, but we should double check, and merge with whatever the test runs were using. @kalverra

**Review by samsondav — APPROVED** [review]
LGTM would be good to dump that old optimism1 estimator if possible.

File: core/chains/evm/config/chain_specific_config.go [hunk 2]
```diff
@@ -332,6 +332,23 @@ func setChainSpecificConfigDefaultSets() {
...
+	metisMainnet.balanceMonitorBlockDelay = 0
+	metisMainnet.blockHistoryEstimatorBlockHistorySize = 0 // Force an error if someone set GAS_UPDATER_ENABLED=true by accident; we never want to run the block history estimator on metis
+	metisMainnet.chainType = config.ChainMetis
+	metisMainnet.finalityDepth = 1    // Sequencer offers absolute finality
+	metisMainnet.gasBumpThreshold = 0 // Never bump gas on metis
+	metisMainnet.gasEstimatorMode = "L2Suggested"
+	metisMainnet.linkContractAddress = "TODO"
+	metisMainnet.minIncomingConfirmations = 1
+	metisMainnet.minGasPriceWei = *big.NewInt(0) // Metis uses the L2Suggested estimator; we don't want to place any limits on the minimum gas price
+	metisMainnet.minRequiredOutgoingConfirmations = 0
```
Comment (jmank88) [inline core/chains/evm/config/chain_specific_config.go:346]: Matches Optimism changes from #6621
```suggestion
	metisMainnet.minRequiredOutgoingConfirmations = 1
```

**Review by samsondav — DISMISSED** [review]

**Comment by Good-w** [comment]
能不能尽快？

**Comment by Good-w** [comment]
How soon can the integration be successful?

File: core/chains/evm/gas/l2_suggested_estimator.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+		make(chan struct{}),
+		make(chan struct{}),
+	}
+}
+
+func (o *l2SuggestedEstimator) Start(context.Context) error {
+	return o.StartOnce("L2SuggestedEstimator", func() error {
+		go o.run()
+		<-o.chInitialised
+		return nil
```
Comment (prashantkumar1982) [inline core/chains/evm/gas/l2_suggested_estimator.go:64]: Shouldn't we check that o.l2GasPrice != nil before returning success here?

If for some reason the rpcClient is unreachable, doesn't that make this estimator useless?

Comment (prashantkumar1982) [inline core/chains/evm/gas/l2_suggested_estimator.go:64]: Shouldn't we check that o.l2GasPrice != nil before returning success here?

If for some reason the rpcClient is unreachable, doesn't that make this estimator useless?

Comment (prashantkumar1982) [reply core/chains/evm/gas/l2_suggested_estimator.go:64]: Oh, looks like we will return error until we are able to get a gas price set by a subsequent refresh.
So don't fail the Start() call, and keep retrying price refresh until it becomes available.

Is that right?

Comment (prashantkumar1982) [reply core/chains/evm/gas/l2_suggested_estimator.go:64]: Oh, looks like we will return error until we are able to get a gas price set by a subsequent refresh.
So don't fail the Start() call, and keep retrying price refresh until it becomes available.

Is that right?

File: core/chains/evm/gas/l2_suggested_estimator.go [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+var (
+	_ Estimator = &l2SuggestedEstimator{}
+)
+
+//go:generate mockery --name rpcClient --output ./mocks/ --case=underscore --structname OptimismRPCClient
+type rpcClient interface {
+	Call(result interface{}, method string, args ...interface{}) error
+}
+
+type l2SuggestedEstimator struct {
```
Comment (prashantkumar1982) [inline core/chains/evm/gas/l2_suggested_estimator.go:26]: nit: Add some documentation here indicating what this estimator does.

Comment (prashantkumar1982) [inline core/chains/evm/gas/l2_suggested_estimator.go:26]: nit: Add some documentation here indicating what this estimator does.

File: core/chains/evm/gas/l2_suggested_estimator.go [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+	gasPriceMu sync.RWMutex
+	l2GasPrice *big.Int
+
+	chForceRefetch chan (chan struct{})
+	chInitialised  chan struct{}
+	chStop         chan struct{}
+	chDone         chan struct{}
+}
+
+// NewL2SuggestedEstimator returns a new optimism 2.0 estimator
```
Comment (prashantkumar1982) [inline core/chains/evm/gas/l2_suggested_estimator.go:43]: nit: remove optimism name from here

Comment (prashantkumar1982) [inline core/chains/evm/gas/l2_suggested_estimator.go:43]: nit: remove optimism name from here

**Review by prashantkumar1982 — APPROVED** [review]
Looks good!

Some minor changes requested, and a couple doubts

File: core/chains/evm/gas/l2_suggested_estimator.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+		make(chan struct{}),
+		make(chan struct{}),
+		make(chan struct{}),
+	}
+}
+
+func (o *l2SuggestedEstimator) Start(context.Context) error {
+	return o.StartOnce("L2SuggestedEstimator", func() error {
+		go o.run()
+		<-o.chInitialised
```
[the remark this answers is not in the document]
Comment (jmank88) [reply core/chains/evm/gas/l2_suggested_estimator.go:63]: That is by design, so `Start` blocks until it gets that signal back from `run()`, which is after first `refreshPrice()` call is complete. It's a pretty standard pattern, but I don't know the importance in this particular case - just renaming for this PR.

**Review by pinebit — APPROVED** [review]

**Comment by jmank88** [comment]
> differences from optimism? (config fields, new client errors to map, etc.)

I am not aware of any differences, but leaving this unchecked since I'm not certain. We can circle back and add anything if we need to.

**Comment by Good-w** [comment]
When will it be done?

**Pull request merged**